### PR TITLE
PR: Remove partial prompt on copy

### PR DESCRIPTION
**Pull request by impact27**
When copying, qtconsole removes whole prompts, but not partial prompts. This removes partial prompts on the first line.

### CODE REVIEW

**Comment by ccordoba12** [comment]
Thanks @impact27 for your contribution. Could you add a test for this one?

**Comment by impact27** [comment]
I had to fix paste as well for the test to pass, there was a corner case

**Comment by ccordoba12** [comment]
Hey @impact27, is this one ready?

**Comment by impact27** [comment]
Yes

File: qtconsole/frontend_widget.py [hunk 1]
```diff
@@ -215,11 +215,54 @@ def copy(self):
...
+
+                # Get first line promp len
+                first_line = preceding_text + first_line_selection
+                len_with_prompt = len(first_line)
+                first_line = remove_prompts(first_line)
+                prompt_len = len_with_prompt - len(first_line)
+
+                # Remove not selected part
+                if prompt_len < len(preceding_text):
+                    first_line = first_line[len(preceding_text)-prompt_len:]
```
Comment (ccordoba12) [inline qtconsole/frontend_widget.py:240]: Please add spaces around the `-` sign.

File: qtconsole/tests/test_jupyter_widget.py [hunk 2]
```diff
@@ -82,3 +82,28 @@ def test_other_output(self):
             '<p style="-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;"><br /></p>\n'
             '<p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;"><span style=" color:#000080;">In [</span><span style=" font-weight:600; color:#000080;">2</span><span style=" color:#000080;">]:</span> </p></body></html>'
         ))
+
+    def test_copy_paste(self):
```
Comment (ccordoba12) [inline qtconsole/tests/test_jupyter_widget.py:86]: Let's rename this to `test_copy_paste_prompt` so that it's clearer what it does.

File: qtconsole/tests/test_jupyter_widget.py [hunk 3]
```diff
@@ -82,3 +82,28 @@ def test_other_output(self):
...
+        cursor.setPosition(pos - 3)
+        cursor.movePosition(QtGui.QTextCursor.End,
+                            QtGui.QTextCursor.KeepAnchor)
+        control.setTextCursor(cursor)
+        control.hasFocus = lambda: True
+        w.copy()
+        clipboard = QtWidgets.QApplication.clipboard()
+        assert clipboard.text() == code
+        w.paste()
+        expected = "In [1]: if True:\n   ...:     print('a')"
```
Comment (ccordoba12) [inline qtconsole/tests/test_jupyter_widget.py:108]: I don't understand this. The text is copied without prompts but when pasted it does show a prompt?

Comment (impact27) [reply qtconsole/tests/test_jupyter_widget.py:108]: Expected is the content of the console. You can't replace the existing prompt when pasting, and a continuation prompt must be created when pasting multilines

Comment (ccordoba12) [reply qtconsole/tests/test_jupyter_widget.py:108]: Ok, I see, thanks for the clarification.

**Review by ccordoba12 — APPROVED** [review]
Thanks @impact27!
